controllers/applicationvolumereplication_controller.go

```diff
@@ -195,295 +191,471 @@ func (r *ApplicationVolumeReplicationReconciler) Reconcile(ctx context.Context,
 
-			continue
+			return nil, fmt.Errorf("failed to get placementrule error: %w", err)
 		}
+	}
 
-		placementDecision, needRequeue := a.processSubscription(&a.subscriptionList.Items[idx])
+	return clonedPlRule, nil
+}
 
-		if needRequeue {
-			a.log.Info("Requeue for subscription", "name", subscription.Name)
+func (r *ApplicationVolumeReplicationReconciler) clonePlacementRule(ctx context.Context,
+	avr *rmn.ApplicationVolumeReplication, userPlRule *plrv1.PlacementRule,
+	clonedPlRuleName string) (*plrv1.PlacementRule, error) {
+	r.Log.Info("Creating a clone placementRule from", "name", userPlRule.Name)
 
-			requeue = true
+	clonedPlRule := &plrv1.PlacementRule{}
 
-			continue
-		}
+	userPlRule.DeepCopyInto(clonedPlRule)
 
-		if a.instance.Status.Decisions == nil {
-			a.instance.Status.Decisions = make(rmn.SubscriptionPlacementDecisionMap)
-		}
+	clonedPlRule.Name = clonedPlRuleName
+	clonedPlRule.ResourceVersion = ""
+	clonedPlRule.Spec.SchedulerName = ""
 
-		if placementDecision != nil {
-			a.instance.Status.Decisions[subscription.Name] = placementDecision
-		}
+	err := r.addClusterPeersToPlacementRule(ctx, avr, clonedPlRule)
+	if err != nil {
+		r.Log.Error(err, "Failed to add cluster peers to cloned placementRule", "name", clonedPlRuleName)
+
+		return nil, err
 	}
 
-	a.log.Info("Returning Placement Decisions", "Total", len(a.instance.Status.Decisions))
+	err = r.Create(ctx, clonedPlRule)
+	if err != nil {
+		r.Log.Error(err, "failed to clone placement rule", "name", clonedPlRule.Name)
 
-	return requeue
+		return nil, errorswrapper.Wrap(err, "failed to coone PlacementRule")
+	}
+
+	return clonedPlRule, nil
 }
 
-func (a *AVRInstance) processSubscription(
-	subscription *subv1.Subscription) (*rmn.SubscriptionPlacementDecision, bool) {
-	a.log.Info("Processing subscription", "name", subscription.Name)
+func (r *ApplicationVolumeReplicationReconciler) addClusterPeersToPlacementRule(ctx context.Context,
+	avr *rmn.ApplicationVolumeReplication, plRule *plrv1.PlacementRule) error {
+	clPeersRef := avr.Spec.DRClusterPeersRef
+	clusterPeers := &rmn.DRClusterPeers{}
 
-	a.log.Info(fmt.Sprintf("processSubscription: current state '%s'",
-		a.instance.Status.LastKnownDRStates[subscription.Name]))
+	if clPeersRef.Namespace == "" {
+		clPeersRef.Namespace = avr.Namespace
+	}
 
-	const requeue = true
-	// Check to see if this subscription is paused for DR. If it is, then restore PVs to the new destination
-	// cluster, unpause the subscription, and skip it until the next reconciler iteration
-	if a.isSubsriptionPausedForDR(subscription.GetLabels()) {
-		a.log.Info("Subscription is paused", "name", subscription.Name)
+	err := r.Client.Get(ctx,
+		types.NamespacedName{Name: clPeersRef.Name, Namespace: clPeersRef.Namespace}, clusterPeers)
+	if err != nil {
+		return fmt.Errorf("failed to get cluster peers using %s/%s. Error (%w)",
+			clPeersRef.Name, clPeersRef.Namespace, err)
+	}
+
+	if len(clusterPeers.Spec.ClusterNames) == 0 {
```

Comment:
(future) Need to determine if ClPeers is setup right? we may not want to do this each time so need to think if ClPeers should be resolved, or let AVR or placement fail. Unsure where to head on this.